lib/gssapi/krb5/unwrap.c

```diff
@@ -175,10 +175,10 @@ unwrap_des
     return GSS_S_BAD_MIC;
   }
 
-  ret = _gssapi_msg_order_check(context_handle->order, seq_number);
-  if (ret) {
+  seq_err = _gssapi_msg_order_check(context_handle->order, seq_number);
+  if (seq_err == GSS_S_FAILURE) {
```

Comment:
I think minor_status needs to be set to 0 before this return.  Otherwise there might be random garbage used by the caller.   We might consider setting *minor_status = 0 upon entry to this function instead of setting it at each error return site.

Reply:
Again, not setting minor_status is a pre-existing issue that isn't affected by using 'seq_err' rather than 'ret' here.